doc/program_services.md

```diff
@@ -0,0 +1,1175 @@
+# Program Services
+
+## Save area and linkage conventions
+
+Standard save area is defined as follows:
+
+``` hlasm
+         DS   0CL72
+         DS   F         +0 reserved
+         DS   F         +4 address of calling savearea
+         DS   F         +8 address of called savearea
+         DS   15F       +12 callers GR14 through GR12
+```
+
+There are normally two save areas involved when a program is called by another program:
+
+* The save area for the calling program -- the caller
+* The save area for the called program -- the callee
+
+Upon entry to the called program, GR13 contains the address of the caller's save area. 
+
+The callee:
+
+1. Saves GR14 to GR12 (registers at entry to the called program) at +12 of the caller's save area
+2. Obtains its own save area (the called program's save area)
+3. Stores the address of its save area at +8 of the caller's save area (caller's forward chain)
+4. Stores the address of the caller's save area (GR13 at entry) at +4 of its (the called program's) save area (called program's back chain)
+5. Loads GR13 with the address of its (the called program's) save area
+
+At this point the called program begins its processing. 
+
+If it calls another program steps 1--5 are repeated; this time the "caller" is the current program and there is a new called program.
+
+When the called program finishes processing, it returns to the caller as follows:
+
+6. The called program uses its back chain (+4 in its save area) to restore GR13 to caller's save area
+7. The called program restores GR14 to GR12 from that save area (restores registers at entry); there are cases when GR15, GR0, and/or GR1 are not restored -- depends on the program. 
+8. The called program returns to the caller by doing an unconditional branch to the address in GR14.
+
+This calling convention is explained in detail in the [IBM z/OS Version 2 Release 4 MVS Programming: Assembler Services
+Guide (SA23-1368-40)](https://www-01.ibm.com/servers/resourcelink/svc00100.nsf/pages/zOSV2R4sa231368/$file/ieaa600_v2r4.pdf). Refer to Chapter 2 - Linkage conventions.
+
+GR15 is expected to contain a return code upon exit by convention.
```

Comment:
This section looks good. You might reword line 43:

"GR15, by convention, may contain a return code upon exit."

Reply:
Fixed by 3f9f2ba